networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/DiscoveryNetwork.java

```diff
@@ -105,6 +107,18 @@ public Optional<String> getEnr() {
     return discoveryService.getEnr();
   }
 
+  @Override
+  public Optional<UInt256> getDiscoveryNodeId() {
+    BigInteger nodeId =
+        new BigInteger(
+            1,
+            discoveryService
+                .getNodeId()
+                .orElseThrow(() -> new IllegalArgumentException("Unable to get discovery node id"))
```

Comment:
we'd better keep optional flow here, why to throw if we have optional return

Reply:
Done